onnx/defs/controlflow/defs.cc

```diff
@@ -5,49 +5,34 @@
 namespace ONNX_NAMESPACE {
 using SupportType = OpSchema::SupportType;
 
-static TypeProto RemoveDimensionsFromShape(
-    const TypeProto& proto,
-    int num_dimensions) {
-  TypeProto t(proto);
-  auto mutable_shape = t.mutable_tensor_type()->mutable_shape();
-  mutable_shape->clear_dim();
-
-  const auto& dims = proto.tensor_type().shape().dim();
-
-  // skip batch size and sequence length dimensions
-  for (int j = num_dimensions, end = dims.size(); j < end; ++j) {
-    (*mutable_shape->add_dim()) = dims.Get(j);
-  }
-
-  return t;
-}
-
-// copied from GSL:
-// https://github.com/Microsoft/GSL/blob/master/include/gsl/gsl_util
-template <class T, class U>
-static constexpr T narrow_cast(U&& u) noexcept {
-  return static_cast<T>(std::forward<U>(u));
-}
-
 void ScanInferenceFunction(InferenceContext& ctx) {
```

Comment:
>ScanInferenceFunction [](start = 5, length = 21)

Would be good to update the unit tests in onnx\test\cpp\shape_inference_test.cc to validate the logic when axis != 0